File: JZOffer_PHP/50_array_duplicate.php
<?php
    运行时间：30ms

    占用内存：2504k
	function duplicate($numbers, &$duplication)
	{
		
		//这里要特别注意~找到任意重复的一个值并赋值到duplication[0]
		//函数返回True/False
		$length = count($numbers);
		if (empty($numbers) || $length <= 0):
			return false;
		
		foreach ($numbers as $item) {
			if ($item < 0 || $item > $length - 1) {
				return false;
			}
		}
		
		for ($i = 0; $i < $length; $i++ ) {
			while ($i != $numbers[$i]) {
				
				if($numbers[$i] == $numbers[$numbers[$i]]){
					$duplication[0] = $numbers[$i];
					return true;
				}
				
				$temp = $numbers[$i];
				$numbers[$i] = $numbers[$temp];
				$numbers[$temp] = $temp;
			}
		}
		
	}
?>

File: JZOffer_Python/50_array_duplicate.py
#!/usr/bin/python

# -*- coding:utf-8 -*-
class Solution:
	def duplicate(self, numbers, duplication):
		# write code here
		if not numbers or len(numbers) <= 0:
			return False
	
		for num in numbers:
			count = 0
			for item in numbers:
				
				if num == item:
					count += 1
				if count > 1:
					duplication[0] = item
					return duplication[0]
					break
		return False
		
#	运行时间：29ms
#
#	占用内存：5736k
	def duplications(self, nums, dups):
		if not nums or len(dups) <= 0:
			return False
		for i in range(len(nums)):
			    if (nums[i] < 0 ) or (nums[i] > len(nums) - 1):
				return False
		for i in range(len(nums)):
			
			while i != nums[i]:
				if nums[i] == nums[nums[i]]:
					dups[0] = nums[i]
					return True
		
				temp = nums[i]
				nums[i] = nums[temp]
				nums[temp] = temp
			
		return False
			
			
s = Solution()
print s.duplicate([2,1,3,0,4], [0])
print s.duplications([2,3,1,0,2,5,3], [0])			